### PR TITLE
WL-3156 Locations and Categories live in harmony

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: tool/pom.xml [hunk 1]
```diff
@@ -128,8 +128,22 @@
 			<version>2.1</version>
 		</dependency>
 
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<version>4.11</version>
```
Comment (buckett) [inline tool/pom.xml:135]: Should have `<scope>test</scope>`, otherwise it ends up in the final JAR.

File: tool/pom.xml [hunk 2]
```diff
@@ -128,8 +128,22 @@
...
+		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<version>1.8.4</version>
```
Comment (buckett) [inline tool/pom.xml:141]: Also missing `<scope>test</scope>`

File: tool/src/java/org/sakaiproject/signup/tool/jsf/SignupMeetingsBean.java [hunk 3]
```diff
@@ -256,15 +256,15 @@ public String removeMeetings() {
 		//avoid multiple calls for one page loading : not refresh within one second
 		if(allCategories == null || curr_time - lastUpdatedCatTime > 1000){
 			List<SelectItem> categories = new ArrayList<SelectItem>();
-			List<String> allCategories = null;
+			List<String> allCats = null;
```
Comment (buckett) [inline tool/src/java/org/sakaiproject/signup/tool/jsf/SignupMeetingsBean.java:259]: Is the rename so you don't need to use this.allCategories?

**Comment by buckett** [comment]
Nice work getting the tests working ( :racehorse: )

File: tool/src/java/org/sakaiproject/signup/tool/jsf/SignupMeetingsBean.java (hunk 3 above)
Comment (ghost) [inline tool/src/java/org/sakaiproject/signup/tool/jsf/SignupMeetingsBean.java:259]: The rename is to make the syntax look like the code in `getAllLocations`, even though I don't particularly agree with it.

`allCategories` is the class variable that holds the cache so I don't think we want to play with that when we are doing intermediary operations. In `getAllLocations` a locally scoped `allLocs` is used instead.

File: tool/pom.xml (hunk 1 above)
Comment (ghost) [inline tool/pom.xml:135]: Done.

**Comment by ghost** [comment]
I think my New Year's Resolution will be right more tests.

**Pull request merged**